File: 문자열/1157 단어 공부.py
########## Python ##########

string = input().upper() #대문자로 변환 후 string에 저장
alp = list(set(string)) #set: 중복 제거

count = []

for i in alp:
    word_cnt = string.count(i) #alp list에 든 글자들을 센다
    count.append(word_cnt) #count한 숫자를 'count' list에 append

if count.count(max(count)) >= 2: #count list에 든 숫자 최댓값 중복 시
    print("?")
else:
    print(alp[count.index(max(count))]) #위치. 즉 인덱스를 출력

File: 문자열/1157 단어 공부.c
////////// C //////////

#define _CRT_SECURE_NO_WARNINGS
#include<stdio.h>

int main() {

    char input[1000000];
    int alp[26] = { 0 };
    int maxalp = 0; //가장 많이 나온 알파벳 번호
    int maxcnt = 0; //count 값
    int twicecnt = 0;

    scanf("%s", &input);

    for (int i = 0; input[i] != 0; i++) {
        if (input[i] < 91) {
            alp[input[i] - 65]++;
        }
        else {
            alp[input[i] - 97]++;
        }
    }
   
    for (int i = 0; i < 26; i++) {
        if (maxcnt < alp[i]) {
            maxcnt = alp[i];
            maxalp = i;
        }
    }

    for (int i = 0; i < 26; i++) {
        if (maxcnt == alp[i]) twicecnt++;
    }


    if (twicecnt >= 2) {
        printf("?");
    }
    else printf("%c", maxalp+65);

    return 0;
}
